src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -262,6 +268,25 @@ def get(self, request: Request, project: Project) -> Response:
                     else:
                         scope.set_tag("stacktrace_link.error", "file_not_found")
 
+                result["commitSha"] = None
+                githubIntegration = GitHubIntegration(integrations[0], project.organization_id)
+                try:
+                    line_no_str = ctx.get("line_no")
+                    line_no = int(line_no_str) if line_no_str else 0
+                    git_blame_list = githubIntegration.get_blame_for_file(
+                        repo=current_config["repository"],
+                        filepath=filepath,
+                        ref=current_config["config"]["defaultBranch"],
+                        lineno=line_no,
+                    )
+
+                    for blame in git_blame_list:
+                        if line_no >= blame["startingLine"] and line_no <= blame["endingLine"]:
+                            result["commitSha"] = blame["commit"]["oid"]
```

Comment:
Set the `commitSha` to the git blame that corresponds with the line number we are looking for